### PR TITLE
INTYGFV-16181: Implementation of getHospCredentialsForPersonResponseType in HsaIntegrationAuthorizationManagementService.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/HsaIntegrationAuthorizationManagementService.java [hunk 1]
```diff
@@ -51,7 +53,13 @@ public List<CredentialInformation> getCredentialInformationForPerson(String pers
 
     @Override
     public HospCredentialsForPerson getHospCredentialsForPersonResponseType(String personalIdentityNumber) {
-        return null;
+        final var credentialsForPerson = getHospCredentialsForPersonService.get(personalIdentityNumber);
```
Comment (mhornfeldt) [inline integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/HsaIntegrationAuthorizationManagementService.java:56]: Should we create a message with the person id without hashing it?

Comment (ChrilleSigerhed) [reply integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/HsaIntegrationAuthorizationManagementService.java]: After reviewing the the usage of this method I decided to remove the log & exception. This was done cause this exception is not handled, and instead, a null check is performed.

```java
 @Override
    public HospPerson getHospPerson(String personId) {

        LOG.debug("Getting hospPerson from HSA for '{}'", personId);

        HospCredentialsForPerson response = null;
        try {
            response = authorizationManagementService.getHospCredentialsForPersonResponseType(personId);
        } catch (SOAPFaultException e) {
            LOG.debug("Soap exception", e);
        }

        if (response == null || response.getPersonalIdentityNumber() == null) {
            LOG.debug("Response did not contain any hospPerson for '{}'", personId);
            return null;
        }
   }
```

File: integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/client/authorization/HsaIntygProxyServiceHospCredentialsForPersonClient.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import se.inera.intyg.infra.integration.intygproxyservice.dto.authorization.GetCredentialsForPersonResponseDTO;
+
+@Service
+public class HsaIntygProxyServiceHospCredentialsForPersonClient {
+
+    @Autowired
+    @Qualifier("hsaIntygProxyServiceRestTemplate")
+    private RestTemplate restTemplate;
+    @Value("${integration.intygproxyservice.credentialsforperson.endpoint}")
+    private String credentialsForPerson;
```
Comment (mhornfeldt) [inline integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/client/authorization/HsaIntygProxyServiceHospCredentialsForPersonClient.java:37]: I saw a comment on another PR that it's preferred to be endpoint at the end of this variable

File: integration/hsa-integration-intyg-proxy-service/src/test/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/HsaIntegrationAuthorizationManagementServiceTest.java [hunk 3]
```diff
@@ -30,15 +30,20 @@
...
 import org.mockito.junit.jupiter.MockitoExtension;
 import se.inera.intyg.infra.integration.hsatk.model.CredentialInformation;
+import se.inera.intyg.infra.integration.hsatk.model.HospCredentialsForPerson;
 import se.inera.intyg.infra.integration.intygproxyservice.dto.authorization.GetCredentialInformationRequestDTO;
 
 @ExtendWith(MockitoExtension.class)
 class HsaIntegrationAuthorizationManagementServiceTest {
 
     private static final String PERSON_HSA_ID = "personHsaId";
+    private static final String PERSON_ID = "personId";
```
Comment (mhornfeldt) [inline integration/hsa-integration-intyg-proxy-service/src/test/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/HsaIntegrationAuthorizationManagementServiceTest.java:40]: Space can be added

**Review by mhornfeldt — APPROVED** [review]

File: integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/GetHospCredentialsForPersonService.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            GetCredentialsForPersonRequestDTO.builder()
+                .personId(personId)
+                .build()
+        );
+        return credentialsForPersonResponseDTO.getCredentials();
+    }
+
+    private void validateRequest(String personId) {
+        if (personId == null || personId.isEmpty()) {
+            throw new IllegalArgumentException("Missing required parameter personId");
```
Comment (mikaelaromild1) [inline integration/hsa-integration-intyg-proxy-service/src/main/java/se/inera/intyg/infra/integration/intygproxyservice/services/authorization/GetHospCredentialsForPersonService.java:46]: Just a suggestion, but should we add apostrophes around the missing parameters for readability purposes? 
For example "Missing required parameter 'personId'"

**Pull request merged**